Lib/test/test_shutil.py

```diff
@@ -33,6 +33,25 @@
 except ImportError:
     UID_GID_SUPPORT = False
 
+
+def _has_functional_lchmod():
+    if not support.can_symlink() or not hasattr(os, 'lchmod'):
+        return False
+
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
A whole temporary directory for a single file seems overkill. Why noy using NamedTemporaryFile instead? Or tempfile.mktemp() and then unlink it?

Reply:
It's a file and a symlink -- then I'd need manual cleanup code (a `try-finally` + `unlink`) -- unless you feel strongly about this I'd rather leave it.  It's in test code after all and ephemeral.

Reply:
Oh actually you're right I can probably get away with just a temp file. Was misremembering my implementation 😆

Reply:
done!